src/core/internal/convert.d

```diff
@@ -597,9 +600,9 @@ private bool isNonReference(T)()
 
 private bool isNonReferenceStruct(T)() if (is(T == struct) || is(T == union))
 {
-    foreach (cur; T.init.tupleof)
+    static foreach (cur; T.tupleof)
     {
-        static if (!isNonReference!(typeof(cur))()) return false;
+        if (!isNonReference!(typeof(cur))()) return false;
```

Comment:
Why do you remove the `static if` here?

Reply:
To avoid the compiler complaining about unreachable code.